strings/base_macros.h

```diff
@@ -15,6 +15,12 @@
 
 #define WINRT_IMPL_SHIM(...) (*(abi_t<__VA_ARGS__>**)&static_cast<__VA_ARGS__ const&>(static_cast<D const&>(*this)))
 
+#ifdef __INTELLISENSE__
```

Comment:
Going through this macro route, it would be a "syntax changing macro" that probably should be included in the [hint file](https://docs.microsoft.com/en-us/cpp/build/reference/hint-files?view=vs-2019).